apps/prairielearn/src/lib/opsbot.ts

```diff
@@ -0,0 +1,72 @@
+import detectMocha from 'detect-mocha';
+
+import { makeWithData } from '@prairielearn/error';
+import { config } from './config';
+import { logger } from '@prairielearn/logger';
+import fetch, { Response } from 'node-fetch';
+
+export function canSendMessages() {
+  return detectMocha() || !!config.secretSlackOpsBotEndpoint;
+}
+
+export async function sendMessage(msg: string): Promise<Response | null> {
+  if (detectMocha()) {
+    return new Response('Dummy test body', { status: 200, statusText: 'OK' });
+  }
+
+  // No-op if there's no url specified
+  if (!config.secretSlackOpsBotEndpoint) {
+    return null;
+  }
+
+  return fetch(config.secretSlackOpsBotEndpoint, {
+    method: 'POST',
+    body: JSON.stringify({ text: msg }),
+    headers: { 'Content-Type': 'application/json' },
+  });
+}
+
+/**
+ * General interface to send a message from PrairieLearn to Slack.
+ * @param msg String message to send.
+ * @param channel Channel to send to.  Private channels must have the bot added.
+ */
+export async function sendSlackMessage(
+  msg: string,
+  channel?: string | null | undefined,
+): Promise<Response | null> {
+  const token = config.secretSlackToken;
+
+  // Log the message if there's no token specified
+  if (!token || !channel) {
+    logger.info(`Slack message:\n${msg}`);
+    return null;
+  }
+
+  if (detectMocha()) {
+    return new Response('Dummy test body', { status: 200, statusText: 'OK' });
+  }
+
+  const response = await fetch('https://slack.com/api/chat.postMessage', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${token}` },
+    body: JSON.stringify({
+      text: msg,
+      channel: channel,
+      as_user: true,
+    }),
+  });
+
+  if (!response.ok) {
+    throw makeWithData(`Error sending message to ${channel}`, { body: await response.json() });
```

Comment:
Conventionally, we import all exports as the `error` object (`import error = require('@prairielearn/error')`) so that we can do `error.makeWithData`, which makes it more obvious _what_ we're actually making.

Eventually, we'll address this TODO so that you can do what you were trying to do here:

https://github.com/PrairieLearn/PrairieLearn/blob/fef8f8a6ec7db9c60a5a6d2711a35cb898d9f78c/packages/error/src/index.ts#L17-L18